File: 13. Number of Ways to Split Array.java
class Solution {
    public int waysToSplitArray(int[] nums) {
        int n = nums.length;
        long totalSum = 0; // Calculate the total sum of the array
        for (int num : nums) {
            totalSum += num;
        }

        long prefixSum = 0; // Initialize the prefix sum
        int validSplits = 0; // Initialize the count of valid splits

        for (int i = 0; i < n - 1; i++) { // Iterate through indices 0 to n-2
            prefixSum += nums[i]; // Update the prefix sum
            if (prefixSum >= totalSum - prefixSum) { // Check the split condition
                validSplits++; // Increment valid split count if condition is satisfied
            }
        }

        return validSplits;
    }

    // Example usage
    public static void main(String[] args) {
        Solution solution = new Solution();
        int[] nums1 = {10, 4, -8, 7};
        int[] nums2 = {2, 3, 1, 0};

        System.out.println(solution.waysToSplitArray(nums1)); // Output: 2
        System.out.println(solution.waysToSplitArray(nums2)); // Output: 2
    }
}

File: 13. Number of Ways to Split Array.py
class Solution(object):
    def waysToSplitArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        n = len(nums)
        total_sum = sum(nums)  # Calculate the total sum of the array
        prefix_sum = 0  # Initialize the prefix sum
        valid_splits = 0  # Initialize the count of valid splits

        for i in range(n - 1):  # Iterate through indices 0 to n-2
            prefix_sum += nums[i]  # Update the prefix sum
            if prefix_sum >= total_sum - prefix_sum:  # Check the split condition
                valid_splits += 1  # Increment valid split count if condition is satisfied

        return valid_splits

# Example usage
nums1 = [10, 4, -8, 7]
nums2 = [2, 3, 1, 0]

solution = Solution()
print(solution.waysToSplitArray(nums1))  # Output: 2
print(solution.waysToSplitArray(nums2))  # Output: 2
